habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java

```diff
@@ -147,6 +160,15 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             log.info("Checking for updates to poetry-monorepo-dependency-plugin...");
             poetryHelper.installPoetryPlugin("poetry-monorepo-dependency-plugin");
         }
+
+        if (StringUtils.isEmpty(username) || StringUtils.isEmpty(password)) {
+            log.info(String.format("Did not find username and password for the <server> with <id> %s. Will pull anonymously", pypiRepoId));
+        }
+        else {
+            log.info(String.format("TESTING CHANGE Adding username and password configuration for %s at %s", pypiRepoId, pypiRepoUrl));
+            poetryHelper.execute(
+                    Arrays.asList("config", String.format("http-basic.%s", pypiRepoId), username, password));
```

Comment:
C: If it's already configured with authentication information we shouldn't overwrite it.

Reply:
On follow-up investigation, the only auth method that poetry currently supports is http-basic auth. So there's no risk that we would be preempting a more secure authentication method by always overwriting it, and by doing so we make it easier for the user to update the auth info.

This feedback can be ignored.